### PR TITLE
Legend Revamp

### PR DESCRIPTION
**Pull request by mwernimont**
Before making a pull request
----------------------------
First . . .
- [x] Clean the code the way Vue likes it - run 'npm run lint --fix'        
- [x] Make sure all tests run

Then check for accessibly compliance
- [ ] Run WAVE plugin 508 compliance tool

Then run Browserstack; check that application works on . . .
- [x] Chrome
- [x] Safari
- [x] Edge
- [x] Firefox
- [x] Samsung Internet
- [x] Internet Explorer 11 (not supported, but still needs at least a working user redirect page)

Finally . . .
- [x] Update the changelog appropriately

Description
-----------
Brand New look and functionality for the legend.  Some decoupling of code and functionality with niffty Vue tidbits.  

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: src/components/LegendModal.vue [hunk 1]
```diff
@@ -0,0 +1,154 @@
+<template>
+  <div id="legendModal">
+    <div id="legendModalExit">
```
Comment (abriggs-usgs) [inline src/components/LegendModal.vue]: Do we want the click event on the icon or on the 'div' containing the icon.

File: src/components/LegendModal.vue [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+</template>
+<script>
+import hruSVG from "../images/states/example.svg"
+export default {
+    name: "LegendModal",
+    components:{
+        hruSVG
+    },
+    methods:{
+        handleClick: function(){
```
Comment (abriggs-usgs) [inline src/components/LegendModal.vue]: cool!!

File: src/components/LegendModal.vue [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+@media screen and (min-width: 600px){
+    #legendModal{
+        max-width: 300px;
+    }
+
+    #legendModalContent{
+        #hruSVG{
+            width: 100%;
+            height: auto;
+        }
```
Comment (abriggs-usgs) [inline src/components/LegendModal.vue]: Sweet formatting on the CSS.

File: src/components/LegendModal.vue (hunk 1 above)
Comment (mwernimont) [reply src/components/LegendModal.vue]: I would say the link is perfect, as the div currently stretches the whole length of the modal, so an adherent click on the left side of the modal would close the info if we went that route.

Comment (mwernimont) [reply src/components/LegendModal.vue]: Which would probably lead to confusion

File: src/components/MapLegend.vue [hunk 4]
```diff
@@ -128,16 +90,24 @@ export default {
...
-          this.$ga.event(eventName, action, label)
-      },
+    runGoogleAnalytics(eventName, action, label) {
+      this.$ga.event(eventName, action, label);
+    },
+    legendToggle(){
+      this.hidden = !this.hidden;
+    },
+    modalToggle(){
+      this.infoShowing = !this.infoShowing;
```
Comment (abriggs-usgs) [inline src/components/MapLegend.vue]: here is a new fun way of doing these toggle -- the ternary statement
modalToggle() {
    this.infoShowing === false ? this.infoShowing = true : this.infoShowing = false
}
Basically it says, if this is false make it true, if it is true make it false. Really handy in combo with v-show.

File: src/components/LegendModal.vue (hunk 1 above)
Comment (abriggs-usgs) [reply src/components/LegendModal.vue]: Totally trust you, dude.

**Pull request merged**